zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/SparkSQLGenerator.scala

```diff
@@ -0,0 +1,636 @@
+        //        println(groupStrs)
+        appendIfNotEmpty(queryBuilder, groupStr)
+
+        group.aggregates.foreach { aggr =>
+          val fieldExpr = exprMap(aggr.field.name)
+          //def
+          val aggrExpr = parseAggregateFunc(aggr, fieldExpr.refExpr)
+          //ref
+          val newExpr = s"$quote${aggr.as.name}$quote"
+          producedExprs += aggr.as.name -> FieldExpr(newExpr, aggrExpr)
+        }
+        if (!group.lookups.isEmpty) {
+          //we need to update producedExprs
+          val producedExprMap = producedExprs.result().toMap
+          val newExprMap =
+            producedExprMap.map {
+              case (field, expr) => field -> FieldExpr(s"$groupedLookupSourceVar.$quote$field$quote", s"$groupedLookupSourceVar.$quote$field$quote")
+            }
+          queryBuilder.insert(0, s"from (\n${parseProject(producedExprMap)}\n")
+          queryBuilder.append(s"\n) $groupedLookupSourceVar\n")
+          val resultAfterLookup = parseLookup(group.lookups, newExprMap, queryBuilder, true)
+          ParsedResult(Seq.empty, resultAfterLookup.exprMap)
+        } else {
+          ParsedResult(Seq.empty, producedExprs.result().toMap)
+        }
+
+
+      case None => ParsedResult(Seq(""), exprMap)
+    }
+  }
+
+
+  private def parseSelect(selectOpt: Option[SelectStatement],
+                          exprMap: Map[String, FieldExpr], query: Query,
+                          queryBuilder: StringBuilder): ParsedResult = {
+    println("exprMap is:", exprMap)
+    selectOpt match {
+      case Some(select) =>
+        val producedExprs = mutable.LinkedHashMap.newBuilder[String, FieldExpr]
+        println("producedExprs is:", producedExprs)
+
+        val orderStrs = select.orderOn.zip(select.order).map {
+          case (orderOn, order) =>
+            val expr = exprMap(orderOn.name).refExpr
+            val orderStr = if (order == SortOrder.DSC) "desc" else ""
+            s"${expr} $orderStr"
+          //          "order by `count` desc": expr = `count`(orderOn), orderStr = desc
+        }
+        println("orderStrs is:", orderStrs)
+
+        val orderStr =
+          if (!orderStrs.isEmpty) {
+            orderStrs.mkString("order by ", ",", "")
+          } else {
+            ""
+          }
+        println("orderStr is:", orderStr)
+
+        val limitStr = s"limit ${select.limit}"
+        appendIfNotEmpty(queryBuilder, orderStr)
+        if (select.limit != 0) {
+          appendIfNotEmpty(queryBuilder, limitStr)
+        }
+
+
+        if (select.fields.isEmpty) {
+          producedExprs ++= exprMap
+          println("producedExprs after ++:", producedExprs)
+        } else {
+          println("exprMap is:", exprMap)
+          select.fields.foreach {
+            case AllField =>
+              producedExprs ++= exprMap
+            case field =>
+              if (field.name == "create_at"){
+                //                producedExprs +=
```

Comment:
"create_at" is not in the FieldExpr, so if variable "field" of SelectStatement class in Query.scala is "create_at", it will give an error